python/PiFinder/camera_none.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/python
+# -*- coding:utf-8 -*-
+"""
+This module is the camera
+* Captures images
+* Places preview images in queue
+* Places solver images in queue
+* Takes full res images on demand
+
+"""
+import os
+import queue
+import time
+from PIL import Image
+from PiFinder import config
+from PiFinder import utils
+from PiFinder.camera_interface import CameraInterface
+from typing import Tuple
+
+
+class CameraNone(CameraInterface):
+    """Simulate a camera not solving"""
+
+    def __init__(self, exposure_time, gain) -> None:
+        self.camera_type = "none"
+        self.camType = f"None {self.camera_type}"
+        self.exposure_time = exposure_time
+        self.gain = gain
+        self.initialize()
+
+    def initialize(self) -> None:
+        """Initializes the camera and set the needed control parameters"""
+        return
+
+    def capture(self) -> Image.Image:
+        return self.camera.capture_image()
+
+    def capture_file(self, filename) -> None:
+        return self.camera.capture_file(filename)
+
+    def set_camera_config(
+        self, exposure_time: float, gain: float
+    ) -> Tuple[float, float]:
+        self.camera.stop()
+        self.camera.set_controls({"AnalogueGain": gain})
+        self.camera.set_controls({"ExposureTime": exposure_time})
+        self.camera.start()
+        return exposure_time, gain
+
+    def get_cam_type(self) -> str:
+        return self.camType
+
+
+def get_images(shared_state, camera_image, command_queue, console_queue):
+    """
+    Instantiates the camera hardware
+    then calls the universal image loop
+    """
```

Comment:
I believe this current implementation will just cause the camera thread to exit.  The `get_images` function is called to start the image loop.  If you copy what is in the camera_debug.py for the `get_images` function, it has the minimum required to start and maintain the image loop.

Then you change the `capture` method of the `CameraNone` object to either return a blank image, a non-solving image from disk, or (my personal favorite) write a random dot generator return plausible, but not solvable, star images for the whole system to chew on.

Reply:
will have another look at this later, but you're right it should do something more than exiting